frontend/src/requests.js

```diff
@@ -50,6 +50,45 @@ export default {
     }
     return await response.json();
   },
+  async getImage(url) {
+    console.log(url);
+    const authToken = authStore.getToken();
+    return await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
```

Comment:
The content-type field caught my eye when looking at how the data is coming back.   https://stackoverflow.com/questions/20508788/do-i-need-content-type-application-octet-stream-for-file-download  When handling files,  returning data as a binary data stream natively saves a lot of effort of turning to and from a file.   The content-disposition https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition, can also be used in order to control whether the file is anticipated to be an 'attachment' to download or a 'inline' image to be displayed. 

We do know what the media type of the file is due to saving its metadata in the attachment information which we have access to. https://en.wikipedia.org/wiki/Media_type also (would not need this for the binary stream, though, cause setting the filename for the attachment, the target system would make the assumption to save as that file type if not inline.)

FastAPI & Starlette sort of supports this, but we may need a custom response in order for the HTTP OpenAPI documentation to make sense.  The https://fastapi.tiangolo.com/advanced/custom-response/#streamingresponse streaming response, expects to just get a file handler and stream reading it to the front, but it doesn't look like pymongo supports streaming the query results of the binary?  So maybe a custom Response that sets the expected content according to the file type being returned might work.  

Take some time to think through the options of what you would like to try and use and lets discuss.

Reply:
The data returned has been updated to include the content type. 

All data, so far, coming back from FastAPI is done via a Gridfs file stream, it is currently using the FastAPI [StreamingResponse](https://fastapi.tiangolo.com/advanced/custom-response/#streamingresponse) which also allows the media type to be set. 

When a file is found in Gridfs, it returns a [GridOut](https://pymongo.readthedocs.io/en/stable/api/gridfs/grid_file.html#gridfs.grid_file.GridOut) object that contains a file iterator that can be handled by FastAPI's StreamingResponse.